### ISSUE TITLE
12$: error on `?` prefix/suffix in type

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#860]
```typescript
fs.promises.writeFile(outputPath, result)                                        
  .then((err: Error?) => {                                                
    if (err) throw err;                                                          
  });                                                                            
```
quick-lint-js should report an error on the `?`, suggesting `| void` instead.

Similar:
```typescript
fs.promises.writeFile(outputPath, result)                                        
  .then((err: ?Error) => {                                                
    if (err) throw err;                                                          
  });                                                                            
```

Note that the following is legal, because `?` has special rules in tuple elements:
```typescript
let f = (a: [string?]) => {};
```

**Comment by AlekLefebvre** [comment]
I'd like to work on this issue.

**Comment by strager** [comment]
I landed this in commit 6fd626258c2996e325080aaaa4963b33e9bb8182.
